third_party/terraform/website/docs/r/dataflow_flex_template_job.html.markdown

```diff
@@ -0,0 +1,59 @@
+---
+subcategory: "Dataflow"
+layout: "google"
+page_title: "Google: google_dataflow_flex_template_job"
+sidebar_current: "docs-google-dataflow-flex-template-job"
+description: |-
+  Creates a job in Dataflow based on a Flex Template.
+---
+
+# google\_dataflow\_flex\_template\_job
+
+Creates a [Flex Template](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates) job on Dataflow, which is an implementation of Apache Beam running on Google Compute Engine. For more information see
+the official documentation for
+[Beam](https://beam.apache.org) and [Dataflow](https://cloud.google.com/dataflow/).
+
+## Example Usage
+
+```hcl
+resource "google_dataflow_flex_template_job" "big_data_job" {
+  provider                = google-beta
+  name                    = "dataflow-flextemplates-job"
+  container_spec_gcs_path = "gs://my-bucket/templates/template.json"
+  parameters = {
+    inputSubscription = "messages"
+  }
+}
+```
+
+[ To Come ...]
+## Note on "destroy" / "apply"
+There are many types of Dataflow jobs.  Some Dataflow jobs run constantly, getting new data from (e.g.) a GCS bucket, and outputting data continuously.  Some jobs process a set amount of data then terminate.  All jobs can fail while running due to programming errors or other issues.  In this way, Dataflow jobs are different from most other Terraform / Google resources.
```

Comment:
Dataflow terminology for these two types of jobs are `streaming` and `batch` jobs respectively.